lua/entities/gmod_wire_expression2/core/sound.lua

```diff
@@ -304,7 +304,7 @@ local function EmitSound(e2, ent, snd, level, pitch, volume)
     end
 
     local emitting_sounds = ent:GetVar("E2_emitting_sounds", {})
-    emitting_sounds[snd] = (emitting_sounds[snd] or 0) + 1
+    emitting_sounds[snd] = math.min((emitting_sounds[snd] or 0) + 1, 32)
```

Comment:
Need to ensure you can't emit sounds past this point as well.

Reply:
I think this limit should be applied on the total number of sounds played, not specifically for each sound.
I could play bird1.wav, bird2.wav, ..., bird9999999.wav and still cause the lagspike no? I guess we should work with a queue to remove the oldest sound played if needed.

Reply:
In which case you should go back to the array of strings.

Reply:
I doubt you can play that many sounds on an entity

Reply:
Yeah

Reply:
Is there a function that returns if a sound is looping? If we can just keep track of the looping sounds playing, we'll know what needs to get cleaned up when the entity is deleted.